groupoff.cc

```diff
@@ -23,16 +26,24 @@ void Groupoff::main(){
   WATCard fulfilledCards[numStudents];
   for (unsigned int i = 0; i < numStudents; i ++) {
     try {
-      _Accept(~Groupoff) break;
+      _Accept(~Groupoff) {
+        // std::cerr<<"groupoff terminatior "<<std::endl;
+        break;
+      }
       _Else {
         yield(groupoffDelay);
         // TODO: ensure waiting = numstudents at start
         int assignCard = mprng(waiting - 1);
+        // std::cerr<<"assign giftcard "<<i<<" to "<<studentIndex[assignCard]<<std::endl;
+        // std::cerr<<"waiting "<<waiting<<std::endl;
         prt.print(Printer::Groupoff, 'D', sodaCost);
         fulfilledCards[i].deposit(sodaCost);
-        giftCards[assignCard].delivery(&fulfilledCards[i]);
-        std::swap(studentIndex[assignCard], studentIndex[waiting]);
+        // std::cerr<<"deposit "<<i<<std::endl;
+        giftCards[studentIndex[assignCard]].delivery(&fulfilledCards[i]);
```

Comment:
im not sure if this makes sense i was trying to assign the card to the person at assignCard

Reply:
assignCard is the index in studentIndex, studentIndex[assignCard] is the index in giftcards

Reply:
before, if u had 2 students, rng could go like:
assigncard = rng(waiting) = rng(1) = 0
then waiting--
assigncard = rng(waiting) = rng(0) = 0 again

so giftcards[assigncard] would be 0 twice, get assigned 2 cards and crash

but now,
assigncard = rng(waiting) = rng(1) = 0
studentindex[assigncard] = 0
giftcard would be assigned, blah blah,
then swap(studentindex[assigncard], studentindex[waiting-1]) would make studentindex[assigncard] = 1
then waiting--
assigncard = rng(waiting) = rng(0) = 0 again
studentindex[assigncard] = 1 this time,
giftcard is assigned to the other student, correct

Reply:
ok i getchu